src/test/osd/RadosModel.h

```diff
@@ -1948,6 +1948,103 @@ class CopyFromOp : public TestOp {
   }
 };
 
+class SetRedirectOp : public TestOp {
+public:
+  string oid, oid_tgt, tgt_pool_name;
+  ObjectDesc src_value;
+  librados::ObjectWriteOperation op;
+  librados::AioCompletion *comp;
+  ceph::shared_ptr<int> in_use;
+  int snap;
+  int done;
+  int r;
+  SetRedirectOp(int n,
+	     RadosTestContext *context,
+	     const string &oid,
+	     const string &oid_tgt,
+	     const string &tgt_pool_name,
+	     TestOpStat *stat)
+    : TestOp(n, context, stat),
+      oid(oid), oid_tgt(oid_tgt), tgt_pool_name(tgt_pool_name),
+      comp(NULL), snap(-1), done(0), 
+      r(0)
+  {}
+
+  void _begin() override
+  {
+    {
+      Mutex::Locker l(context->state_lock);
+      context->oid_in_use.insert(oid);
+      context->oid_not_in_use.erase(oid);
+      context->oid_in_use.insert(oid_tgt);
+      context->oid_not_in_use.erase(oid_tgt);
+
+      // choose source snap
+      if (0 && !(rand() % 4) && !context->snaps.empty()) {
```

Comment:
we can remove this branch since it's disabled.  and it's fine i think to ignore redirects to snaps.